### DISCUSSION TITLE
IOTDevice Error

### DISCUSSION THREAD

**Discussion by tangytang** [iTwin/community#38] [Q&A]
## Description
We are trying to deploy the [itwin/iot-demo](https://github.com/iTwin/iot-demo) on AWS, but are running into some issues.

Visualizer hang after login (model is displayed but it is not possible to click anything , UI does not react on clicks). 

## Logs

The following error appears in the console:

```
BE_SQLITE_ERROR: ECClass 'IoTdevice.devicepointobservesspatialelement' does not exist or could not be loaded. at DbQueryError.throwIfError (http://localhost:3000/static/js/vendors~main.chunk.js:267382:13) at ECSqlReader.runWithRetry (http://localhost:3000/static/js/vendors~main.chunk.js:268949:67) at async ECSqlReader.readRows (http://localhost:3000/static/js/vendors~main.chunk.js:268923:18) at async ECSqlReader.fetchRows (http://localhost:3000/static/js/vendors~main.chunk.js:268989:23) at async ECSqlReader.step (http://localhost:3000/static/js/vendors~main.chunk.js:269002:7) at async CheckpointConnection.query (http://localhost:3000/static/js/vendors~main.chunk.js:308957:12) at async push../src/IoTExtension/clients/ITwinLink.ts.ITwinLink.fetchDevices (http://localhost:3000/static/js/m
```

The code section can be found in iTwinLinks.ts:

```
private static executeQuery = async (query: string) => {
    const rows = [];
    try {
      for await (const row of ITwinLink.iModel.query(query, undefined, { rowFormat: QueryRowFormat.UseJsPropertyNames }))
        rows.push(row);
    } catch (error) {
      console.error(error);
    }

    return rows;
  };
```

**Comment by SheetalNair** [comment]
@tangytang ,We have used a demo schema to provision our imodel and added some devices for testing, and the **fetchDevices** function is basically retriving that information from the iModel. As you do not have that provisioned iModel, you can replace the existing code and make the **fetchDevices** method return a hardcoded array of objects which looks like this
[{
    label: "Concourse-Room1-T1",
    id: "0xb3000000003e",
    origin: {
        "x": 47.471548240841656,
        "y": 29.16899822181937,
        "z": -2.8318770791543475
    },
    type: "TC-Room-Temperature-Sensor",
    phenomenon: "Temperature",
    iotId: "iothub:device:Concourse-Room1-T1",
    observedElement: "0x200000010db"
}]

Where,
id: element id of the sensor in the imodel
label: name of the sensor
origin: Point3D coordinates denoting the location of the sensor in the imodel
type: Device type
iotId: The id of the sensor shown in AWS
phenomenon: phenomenon the sensor measures 
observedElement: element id of the area monitored by the sensor in the iModel

Note: If you do not know what should be there for id,observedElement or origin value, you can use the same values I have provided in the above sample object and use a Bentley metrostation model and select Concourse level from the widget. The iotId property can be the id of your AWS thing if you have an existing device in AWS iot core.
